addon/services/session.js

```diff
@@ -11,5 +18,80 @@ export default SessionServiceESA.extend({
 
     // Trigger a single logout on the authorization server
     return authenticator.singleLogout(idToken);
-  },
-});
+  }
+
+  get redirectUri() {
+    const { protocol, host } = location;
+    const path = this.router.currentURL;
+    return `${protocol}//${host}${path}`;
+  }
+
+  /**
+   * Watch the `data.authenticated.id_token` to recomputed the headers as
+   * according to the openid-connect specification the `id_token` must always
+   * be included.
+   * https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse
+   */
+  get headers() {
+    const headers = {};
+
+    if (this.isAuthenticated) {
+      const token = this.data.authenticated[tokenPropertyName];
+      Object.assign(headers, { [authHeaderName]: `${authPrefix} ${token}` });
+    }
+
+    return headers;
+  }
+
+  @enqueueTask
+  *refreshAuthentication() {
+    const expireTime = this.data.authenticated.expireTime;
+    const isExpired = expireTime && expireTime <= new Date().getTime();
```

Comment:
refreshLeeway

Reply:
The `refreshLeeway` is already subtracted from the `expireTime` in `this.data.authenticated.expireTime` during authentication and refresh.